# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Marking.cpp
# Commit: 453aa2faad97
# Full Hash: 453aa2faad97ce68b33bef75cd232e37d759556e
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-10-29 09:50:10
# Regressor Bug: 1797755
# File Overlap Count: 1
# Description:
#   Bug 1797755 - Part 4: Remove option to set maximum mark stack capacity in release builds r=sfink
#   
#   Currently we have a GC parameter that allows setting a maximum mark stack
#   capacity. This is only ever used by test code, not in the browser. This
#   requires extra unnecessary work in release builds if we move to a different
# ==============================================================================

diff -r d33708e4f8ec -r 453aa2faad97 js/src/gc/Marking.cpp
--- a/js/src/gc/Marking.cpp	Fri Oct 28 15:17:43 2022 +0000
+++ b/js/src/gc/Marking.cpp	Fri Oct 28 15:17:43 2022 +0000
@@ -1562,15 +1562,7 @@
   return ptr_;
 }
 
-MarkStack::MarkStack(size_t maxCapacity)
-    : topIndex_(0),
-      maxCapacity_(maxCapacity)
-#ifdef DEBUG
-      ,
-      iteratorCount_(0)
-#endif
-{
-}
+MarkStack::MarkStack() { MOZ_ASSERT(isEmpty()); }
 
 MarkStack::~MarkStack() {
   MOZ_ASSERT(isEmpty());
@@ -1590,13 +1582,14 @@
     capacity = NON_INCREMENTAL_MARK_STACK_BASE_CAPACITY;
   }
 
-  if (capacity > maxCapacity_) {
-    capacity = maxCapacity_;
-  }
+#ifdef JS_GC_ZEAL
+  capacity = std::min(capacity, maxCapacity_.ref());
+#endif
 
   return resize(capacity);
 }
 
+#ifdef JS_GC_ZEAL
 void MarkStack::setMaxCapacity(size_t maxCapacity) {
   MOZ_ASSERT(maxCapacity != 0);
   MOZ_ASSERT(isEmpty());
@@ -1608,6 +1601,7 @@
     (void)resize(maxCapacity_);
   }
 }
+#endif
 
 inline MarkStack::TaggedPtr* MarkStack::topPtr() { return &stack()[topIndex_]; }
 
@@ -1686,7 +1680,12 @@
 }
 
 MOZ_NEVER_INLINE bool MarkStack::enlarge(size_t count) {
-  size_t newCapacity = std::min(maxCapacity_.ref(), capacity() * 2);
+  size_t newCapacity = capacity() * 2;
+
+#ifdef JS_GC_ZEAL
+  newCapacity = std::min(newCapacity, maxCapacity_.ref());
+#endif
+
   if (newCapacity < capacity() + count) {
     return false;
   }